### PR TITLE
Add sts web identity provider and process provider to default chain

### PR DESCRIPTION
**Pull request by singku**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/credentials_provider_default_chain.c [hunk 1]
```diff
@@ -279,20 +282,36 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
...
+    struct aws_credentials_provider_profile_options profile_options = {
+        .shutdown_options = sub_provider_shutdown_options, .bootstrap = options->bootstrap};
     profile_provider = aws_credentials_provider_new_profile(allocator, &profile_options);
     if (profile_provider != NULL) {
         providers[index++] = profile_provider;
         /* 1 shutdown call from the profile provider's shutdown */
         aws_atomic_fetch_add(&impl->shutdowns_remaining, 1);
     }
 
+    struct aws_credentials_provider_process_options process_options = {.shutdown_options =
```
Comment (singku) [inline source/credentials_provider_default_chain.c:305]: This disgusting format is done by clang-format

File: source/credentials_provider_default_chain.c [hunk 2]
```diff
@@ -279,20 +282,36 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
...
     }
-
     providers[index++] = environment_provider;
 
-    struct aws_credentials_provider_profile_options profile_options;
-    AWS_ZERO_STRUCT(profile_options);
-    profile_options.bootstrap = options->bootstrap;
-    profile_options.shutdown_options = sub_provider_shutdown_options;
+    struct aws_credentials_provider_sts_web_identity_options sts_web_identity_options = {
+        .shutdown_options = sub_provider_shutdown_options, .bootstrap = options->bootstrap};
```
Comment (bretambrose) [inline source/credentials_provider_default_chain.c:288]: put a comma after bootstrap

File: source/credentials_provider_default_chain.c [hunk 3]
```diff
@@ -279,20 +282,36 @@ struct aws_credentials_provider *aws_credentials_provider_new_chain_default(
...
+
+    sts_web_identity_provider = aws_credentials_provider_new_sts_web_identity(allocator, &sts_web_identity_options);
+    if (sts_web_identity_provider != NULL) {
+        providers[index++] = sts_web_identity_provider;
+        aws_atomic_fetch_add(&impl->shutdowns_remaining, 1);
+    }
+
+    struct aws_credentials_provider_profile_options profile_options = {
+        .shutdown_options = sub_provider_shutdown_options, .bootstrap = options->bootstrap};
```
Comment (bretambrose) [inline source/credentials_provider_default_chain.c:297]: same here

File: source/credentials_provider_default_chain.c (hunk 1 above)
Comment (bretambrose) [reply source/credentials_provider_default_chain.c:305]: comma after sub_provider_shutdown_options

**Review by bretambrose — COMMENTED** [review]
I also think that some of the ERROR log lines (in s_parameters_new) should become WARN or INFO appropriately, otherwise every log will have ERROR lines in it despite the customer probably not wanting web identity credentials.

**Review by bretambrose — COMMENTED** [review]
I meant to suggest, and I don't know if it's worth doing, but at least consider and make your own decision: is there a way to make the log level conditional - ERROR when the provider has been manually set and WARN when it is being used as part of the default chain?  The config struct is one way you could do this.

**Comment by singku** [comment]
Probably we could but the immediate idea would make logic ugly.

**Pull request closed without merging**